include/frg/vector.hpp

```diff
@@ -20,7 +23,13 @@ class vector {
 
 	vector(Allocator &allocator);
 
-	vector(const vector &other) = delete;
+        vector(const vector &other)
+        : vector(*other._allocator) {
+                auto other_size = other.size();
+                _ensure_capacity(other_size);
+                memcpy(_elements, other.data(), other_size * sizeof(T));
```

Comment:
This will break for non-trivially-copyable types. In the general case, we should use placement new to call the copy constructor in a loop.